### PR TITLE
Add memcheck

### PR DESCRIPTION
**Pull request by zanebeckwith**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
+
+  - cd memcheck
+  - cmake --build .
+  - ctest -V -T memcheck
+  - cd ..
+
 after_success:
   - cd debug-with-coverage
   - ../../.travis/upload-gcov-results.sh `pwd`
+  - ../../.travis/show-memcheck-results.sh
```
Comment (drbild) [inline .travis.yml]: Missing the parameter to the script specifying the build dir.  I guess just `pwd` like gcov results.

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,10 @@ SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wextra -std=c99")
 
 SET(CMAKE_C_FLAGS_DEBUGWITHCOVERAGE "${CMAKE_C_FLAGS_DEBUGWITHCOVERAGE} -O0 -fprofile-arcs -ftest-coverage")
 
+find_program(MEMORYCHECK_COMMAND NAMES valgrind)
+set(MEMORYCHECK_COMMAND_OPTIONS
+        "--error-exitcode=5 --leak-check=full -v --track-origins=yes --partial-loads-ok=no")
```
Comment (drbild) [inline CMakeLists.txt]: Could you add a short section to the readme indicating what valgrind is checking.  For folks who don't speak valgrind options, what sorts of errors, roughly, do we intend valgrind to catch?

File: test/sign-and-verify-tests.c [hunk 3]
```diff
@@ -53,12 +53,14 @@ typedef struct sign_and_verify_fixture {
...
 static void teardown(sign_and_verify_fixture* fixture);
 
-int main()
+int main(int argc, char* argv[])
 {
     sign_then_verify_good();
     sign_then_verify_on_rev_list();
-    sign_benchmark();
-    verify_benchmark();
+    if (argc > 1 && 0 == strcmp(argv[1], "benchmark")) {
```
Comment (drbild) [inline test/sign-and-verify-tests.c]: Maybe split the benchmarks into a separate file and exclude that from the `ctest -T memcheck` using the `-E` flag?

File: .travis.yml (hunk 1 above)
Comment (zanebeckwith) [reply .travis.yml]: Thanks, good catch. That got lost in a rebasing, and for some reason travis didn't complain when this script exited with 1.

File: CMakeLists.txt (hunk 2 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: Good call.

File: test/sign-and-verify-tests.c (hunk 3 above)
Comment (zanebeckwith) [reply test/sign-and-verify-tests.c]: Good call.

File: .travis.yml (hunk 1 above)
Comment (zanebeckwith) [reply .travis.yml]: Ah, I didn't realize that only in the `script` stage does a non-zero return code fail the build. The `after_success` doesn't care what the return codes are (makes sense, this is "after success").

File: CMakeLists.txt (hunk 2 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: Good idea. Done.

**Pull request merged**